### PR TITLE
Remove the synthetic preview

### PR DESCRIPTION
**Pull request by 0sewa0**
## Description

Removes the code related to synthetic.

Also "fixes" some of the scripts that are related to docs generation, pls check those carefully: [here](https://github.com/Dynatrace/dynatrace-operator/commit/245f16d5e44da6e1c1651dba3a6f92220686655d)

## How can this be tested?

Check that there are no remaining synthetic code/references/docs in the code.

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
@0sewa0 could you please run `make doc/api-ref`?

File: doc/api/feature-flags.md [hunk 1]
```diff
@@ -25,32 +25,17 @@ const (
 
     // Deprecated: AnnotationFeatureDisableActiveGateUpdates use AnnotationFeatureActiveGateUpdates instead
     AnnotationFeatureDisableActiveGateUpdates = AnnotationFeaturePrefix + "disable-activegate-updates"
-    // Deprecated: AnnotationFeatureDisableActiveGateRawImage use AnnotationFeatureActiveGateRawImage instead
```
Comment (luhi-DT) [inline doc/api/feature-flags.md:28]: why was this removed

File: doc/api/feature-flags.md [hunk 2]
```diff
@@ -25,32 +25,17 @@ const (
...
-    AnnotationFeatureDisableActiveGateRawImage = AnnotationFeaturePrefix + "disable-activegate-raw-image"
 
-    AnnotationFeatureActiveGateUpdates   = AnnotationFeaturePrefix + "activegate-updates"
-    AnnotationFeatureActiveGateRawImage  = AnnotationFeaturePrefix + "activegate-raw-image"
```
Comment (luhi-DT) [inline doc/api/feature-flags.md:32]: and this

File: doc/api/feature-flags.md [hunk 3]
```diff
@@ -25,32 +25,17 @@ const (
...
-    // Deprecated: AnnotationFeatureDisableHostsRequests use AnnotationFeatureHostsRequests instead
-    AnnotationFeatureDisableHostsRequests = AnnotationFeaturePrefix + "disable-hosts-requests"
-    AnnotationFeatureHostsRequests        = AnnotationFeaturePrefix + "hosts-requests"
-    AnnotationFeatureNoProxy              = AnnotationFeaturePrefix + "no-proxy"
-    AnnotationFeatureApiRequestThreshold  = AnnotationFeaturePrefix + "dynatrace-api-request-threshold"
-
-    // Deprecated: AnnotationFeatureDisableReadOnlyOneAgent use AnnotationFeatureReadOnlyOneAgent instead
-    AnnotationFeatureDisableReadOnlyOneAgent = AnnotationFeaturePrefix + "disable-oneagent-readonly-host-fs"
-
-    AnnotationFeatureReadOnlyOneAgent = AnnotationFeaturePrefix + "oneagent-readonly-host-fs"
```
Comment (luhi-DT) [inline doc/api/feature-flags.md:53]: and this

File: doc/api/feature-flags.md [hunk 4]
```diff
@@ -59,13 +44,10 @@ const (
     AnnotationFeatureRunOneAgentContainerPrivileged = AnnotationFeaturePrefix + "oneagent-privileged"
     AnnotationFeatureOneAgentSecCompProfile         = AnnotationFeaturePrefix + "oneagent-seccomp-profile"
 
-    // Deprecated: AnnotationFeatureDisableWebhookReinvocationPolicy use AnnotationFeatureWebhookReinvocationPolicy instead
-    AnnotationFeatureDisableWebhookReinvocationPolicy = AnnotationFeaturePrefix + "disable-webhook-reinvocation-policy"
     // Deprecated: AnnotationFeatureDisableMetadataEnrichment use AnnotationFeatureMetadataEnrichment instead
     AnnotationFeatureDisableMetadataEnrichment = AnnotationFeaturePrefix + "disable-metadata-enrichment"
 
-    AnnotationFeatureWebhookReinvocationPolicy = AnnotationFeaturePrefix + "webhook-reinvocation-policy"
```
Comment (luhi-DT) [inline doc/api/feature-flags.md:67]: and this

**Review by luhi-DT — CHANGES_REQUESTED** [review]

File: doc/api/feature-flags.md (hunk 2 above)
Comment (0sewa0) [reply doc/api/feature-flags.md:32]: its not used

File: doc/api/feature-flags.md (hunk 3 above)
Comment (0sewa0) [reply doc/api/feature-flags.md:53]: its not used

File: doc/api/feature-flags.md (hunk 4 above)
Comment (0sewa0) [reply doc/api/feature-flags.md:67]: its not used

**Review by albertogdd — APPROVED** [review]
LGTM

**Review by luhi-DT — APPROVED** [review]

**Pull request merged**